pkg/isoeditor/nmstate_handler.go

```diff
@@ -0,0 +1,134 @@
+package isoeditor
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+//go:generate mockgen -package=isoeditor -destination=mock_nmstate_handler.go . NmstateHandler
+type NmstateHandler interface {
+	CreateNmstateRamDisk(rootfsPath, ramDiskPath string) error
+}
+
+type nmstateHandler struct {
+	workDir  string
+	executer Executer
+}
+
+func NewNmstateHandler(workDir string, executer Executer) NmstateHandler {
+	return &nmstateHandler{
+		workDir:  workDir,
+		executer: executer,
+	}
+}
+
+func (n *nmstateHandler) CreateNmstateRamDisk(rootfsPath, ramDiskPath string) error {
+	// Extract nmstatectl binary
+	var err error
+	nmstateDir := filepath.Join(n.workDir, "nmstate")
+	err = os.MkdirAll(nmstateDir, os.ModePerm)
+	if err != nil {
+		return err
+	}
+	binaryPath, err := n.extractNmstatectl(rootfsPath, nmstateDir)
+	if err != nil {
+		return err
+	}
+	nmstatectlPath := filepath.Join(nmstateDir, "squashfs-root", binaryPath)
+
+	// Check if nmstatectl binary file exists
+	if _, err = os.Stat(nmstatectlPath); os.IsNotExist(err) {
+		return err
+	}
+
+	// Read binary
+	nmstateBinContent, err := os.ReadFile(nmstatectlPath)
+	if err != nil {
+		return err
+	}
+
+	// Create a compressed RAM disk image with the nmstatectl binary
+	compressedCpio, err := generateCompressedCPIO(nmstateBinContent, NmstatectlPathInRamdisk, 0o100_755)
+	if err != nil {
+		return err
+	}
+
+	// Write RAM disk file
+	err = os.WriteFile(ramDiskPath, compressedCpio, 0755) //nolint:gosec
+	if err != nil {
+		return err
+	}
+
+	// Remove temp dir
+	defer func() {
+		removeErr := os.RemoveAll(nmstateDir)
+		if removeErr != nil {
+			log.WithError(removeErr).Error("failed to remove nmstate temp dir")
+		}
+	}()
+
+	return err
+}
+
+func (n *nmstateHandler) extractNmstatectl(rootfsPath, nmstateDir string) (string, error) {
+	_, err := n.executer.Execute(fmt.Sprintf("7z x %s", rootfsPath), nmstateDir)
```

Comment:
Any reason we picked system's 7z over native golang bindings?

Reply:
I faced an issue when trying to decompress the rootfs due to a format problem that only worked with 7z. Together with Daniel, I tried several Go libraries, but none of them worked for us..

Reply:
@eifrach